src/Common/MemoryTracker.cpp

```diff
@@ -151,6 +155,7 @@ void MemoryTracker::alloc(Int64 size)
             << " (attempt to allocate chunk of " << size << " bytes)"
             << ", maximum: " << formatReadableSizeWithBinarySuffix(current_hard_limit);
 
+        amount.fetch_sub(size, std::memory_order_relaxed);
```

Comment:
free had been replaced with subtraction from amount for proper accounting in parent tracker?

Reply:
Yes (I commented it in this PR above), otherwise we erroneously subtract from parent, but we did not add anything yet.